runtime/compiler/env/VMJ9.cpp

```diff
@@ -7488,120 +7488,110 @@ TR_J9VM::inlineNativeCall(TR::Compilation * comp, TR::TreeTop * callNodeTreeTop,
          // 1      0      -1
          // fall through intended
          return 0;   // FIXME:: disabled for now
+
       case TR::java_lang_invoke_MethodHandles_getStackClass:
+         return 0;
+
       case TR::sun_reflect_Reflection_getCallerClass:
-         // A() -> B() -> s/r/R.getCallerClass(1) => B()
-         // 2      1      0
          {
-         //we need to bail out since we create a class pointer const with cpIndex of -1
-         if ((isAOT_DEPRECATED_DO_NOT_USE() && !comp->getOption(TR_UseSymbolValidationManager)) ||
-             ((!(vmThread()->javaVM->extendedRuntimeFlags &  J9_EXTENDED_RUNTIME_ALLOW_GET_CALLER_CLASS)) &&
-              methodID == TR::sun_reflect_Reflection_getCallerClass))
+         // We need to bail out since we create a class pointer const with cpIndex of -1
+         if (isAOT_DEPRECATED_DO_NOT_USE() && !comp->getOption(TR_UseSymbolValidationManager))
             {
             return 0;
             }
 
-         // char *caseName = (methodID == TR::sun_reflect_Reflection_getCallerClass) ? "s/r/R.getCallerClass" : "j/l/CL.getStackClassLoader";
-         if (callNode->getNumChildren() == 0 || callNode->getFirstChild()->getOpCodeValue() == TR::iconst)
+         static const bool disableGetCallerClassReduction = feGetEnv("TR_DisableGetCallerClassReduction") != NULL;
+         if (disableGetCallerClassReduction)
             {
-            int32_t stackDepth = callNode->getNumChildren() == 0 ? -1 : callNode->getFirstChild()->getInt();
-            if (stackDepth <= 0)
-               return 0;   // getCallerClass is meaningless when invoked with depth <= 0
-            int32_t callerIndex = callNode->getByteCodeInfo().getCallerIndex();
-            int32_t inlineDepth;
-            J9Class *callerClass;
-            J9Method *callerMethod;
-
-            if (methodID == TR::sun_reflect_Reflection_getCallerClass)
-               inlineDepth = 1;
-            else
-               inlineDepth = 0;   // traverse one frame more than getCallerClass
+            return 0;
+            }
+            
+         int32_t targetInlineDepth = 1;
+         
+         int32_t callerIndex = callNode->getByteCodeInfo().getCallerIndex();
+         J9Class *callerClass = NULL;
+         J9Method *callerMethod = NULL;
+
+         // We need to keep track of whether we are asked to skip the frame at depth 0 which is the caller of the
+         // method that calls getStackClass(). We could encounter the case where we have reached caller index -1
```

Comment:
Was this comment meant to say `method that calls getCallerClass` instead ?

Reply:
Fixed in [force-push](https://github.com/eclipse-openj9/openj9/compare/026cc75adea21e664a674b7437e87278a0c0a516..f370b1b2c769e5ac2cb8f27f58dfb020404408b7).